sensors/git_polling_sensor.py

```diff
@@ -0,0 +1,201 @@
+import six
+import eventlet
+from github import Github
+
+import json
+from st2client.client import Client
+from st2client.models import KeyValuePair
+
+from st2reactor.sensor.base import PollingSensor
+
+eventlet.monkey_patch(
+    os=True,
+    select=True,
+    socket=True,
+    thread=True,
+    time=True)
+
+DATE_FORMAT_STRING = '%Y-%m-%d %H:%M:%S'
+
+
+# Default Github API url
+DEFAULT_API_URL = 'https://api.github.com'
+
+
+class GitPollingSensor(PollingSensor):
+    def __init__(self, sensor_service, config=None, poll_interval=None):
+        super(GitPollingSensor, self).__init__(sensor_service=sensor_service,
+                                                     config=config,
+                                                     poll_interval=poll_interval)
+        self._trigger_ref = 'github.repository_activity'
+        self._logger = self._sensor_service.get_logger(__name__)
+
+        self._last_event_ids = {}
+        self.EVENT_TYPE_WHITELIST = []
+        self._st2client = Client()
+        self._repositories_cache = {}
+
+    def setup(self):
+        activity_sensor = self._config.get('activity_sensor', None)
+        if activity_sensor is None:
+            raise ValueError('"activity_sensor" config value is required.')
+
+        self.EVENT_TYPE_WHITELIST = activity_sensor.get('event_type_whitelist', [])
+
+    def poll(self):
+        self._logger.debug('Checking repository activity')
+        gitorgs = self._st2client.keys.get_by_name(name='git-orgs', decrypt=True)
+        if gitorgs:
+            orgs=json.loads(gitorgs.value)
+        else:
+            orgs={}
+        for org_name in orgs:
+            org = orgs[org_name]
+            if 'repositories' not in org:
+                continue
+            token = org['token']
+            user = org['user']
+            repositories = org['repositories']
+            event_type_whitelist = org.get('event_type_whitelist', self.EVENT_TYPE_WHITELIST)
+
+            if org['type'] == 'online':
+                config_base_url = DEFAULT_API_URL
+            else:
+                if org['url']:
+                    config_base_url = org['url']
+                else:
+                    self._logger.debug('URL is not configured for org: "%s"' % (user))
+                    continue
+                
+            client = Github(token or None, base_url=config_base_url)
+            user = client.get_user(user)
+            for repo_name in repositories:
+                repository = {}
+                if org_name + ':' + repo_name + ':' + config_base_url in self._repositories_cache:
```

Comment:
Previously (in `registry_org`), `|` was used to separate the url part

Reply:
I replace it.